pkg/server/api_buildimage.go

```diff
@@ -126,21 +142,46 @@ func (i *InventoryServer) apiBuildImagePostStart(resp http.ResponseWriter, req *
 				}
 				return nil
 			}
+			logrus.Debugf("build-image: build job already running with token %s and seed image %s.", sanitizedJob.Token, reg.buildImageURL)
 			errMsg := fmt.Errorf("a build image task is already running for the MachineRegistration")
 			http.Error(resp, errMsg.Error(), http.StatusBadRequest)
 			return errMsg
 		}
 	}
 
+	// Pod lifecycle: currently we allow just one build task at a time that stays around till we don't start another one.
+	// TODO: extend to allow multiple build Pods
+	// TODO: ensure proper clean-up when the operator is removed or upgraded
+	regList := i.registrationCache.getRegistrationKeys()
+	for _, token := range regList {
```

Comment:
We want to allow just 1 build at a time for now, so we have to check there are no running, otherwise we fail the new job. A new job is allowed only if the previous one completed or failed.